Makefile

```diff
@@ -1,143 +1,89 @@
-	mksys -t wp76xx $(MKSYS_ARGS_COMMON) $(MKSYS_ARGS_RED) mangOH.sdef
-
-.PHONY: red_wp77xx
-red_wp77xx: legato_wp77xx
-	TOOLCHAIN_DIR=$(shell $(LEGATO_ROOT)/bin/findtoolchain wp77xx dir) \
-	TOOLCHAIN_PREFIX=$(shell $(LEGATO_ROOT)/bin/findtoolchain wp77xx prefix) \
-	MANGOH_BOARD=red \
-	mksys -t wp77xx $(MKSYS_ARGS_COMMON) $(MKSYS_ARGS_RED) mangOH.sdef
+.PHONY: all $(ALL_COMBINATIONS)
+all: $(ALL_COMBINATIONS)
 
 # Until Legato allows external builds in mdefs we will need to build a
-# subsystem driver like Cypress Wifi in the top makefile and with scripts
-# Define a function to build the Cypress driver for different architectures
+# subsystem driver like Cypress Wifi in the top Makefile and with scripts
+# define a function to build the Cypress driver for different architectures
 
-# $(call cyp_bld,build_dir)
+# $(call cyp_bld,$(board)_$(module))
+# E.g., $(call cyp_bld,yellow_wp76xx)
 define cyp_bld
-	if [ ! -d $(MANGOH_ROOT)/build/$1/modules/cypwifi ] ; then \
-		mkdir -p $(MANGOH_ROOT)/build/$1/modules/cypwifi ;\
-		cp -pr $(MANGOH_ROOT)/linux_kernel_modules/cypwifi $(MANGOH_ROOT)/build/$1/modules/ ; \
-		$(MANGOH_ROOT)/build/$1/modules/cypwifi/build.sh $1 clean; \
+	if [ ! -d build/$1/modules/cypwifi ] ; then \
+		mkdir -p build/$1/modules/cypwifi ;\
+		cp -pr linux_kernel_modules/cypwifi build/$1/modules/ ; \
+		build/$1/modules/cypwifi/build.sh $1 clean; \
 	else \
-		$(MANGOH_ROOT)/build/$1/modules/cypwifi/build.sh $1 modules; \
+		build/$1/modules/cypwifi/build.sh $1 modules; \
 	fi
 endef
 
-.PHONY: yellow_wp85
-yellow_wp85: legato_wp85
-	$(call cyp_bld,$@)
-	TOOLCHAIN_DIR=$(shell $(LEGATO_ROOT)/bin/findtoolchain wp85 dir) \
-	TOOLCHAIN_PREFIX=$(shell $(LEGATO_ROOT)/bin/findtoolchain wp85 prefix) \
-	MANGOH_BOARD=yellow \
-	mksys -t wp85 $(MKSYS_ARGS_COMMON) $(MKSYS_ARGS_YELLOW) mangOH.sdef
-
-.PHONY: yellow_wp750x
-yellow_wp750x: legato_wp750x
-	$(call cyp_bld,$@)
-	TOOLCHAIN_PREFIX=$(shell $(LEGATO_ROOT)/bin/findtoolchain wp750x prefix) \
-	MANGOH_BOARD=yellow \
-	mksys -t wp750x $(MKSYS_ARGS_COMMON) $(MKSYS_ARGS_YELLOW) mangOH.sdef
-
-.PHONY: yellow_wp76xx
-yellow_wp76xx: legato_wp76xx
-	$(call cyp_bld,$@)
-	TOOLCHAIN_DIR=$(shell $(LEGATO_ROOT)/bin/findtoolchain wp76xx dir) \
-	TOOLCHAIN_PREFIX=$(shell $(LEGATO_ROOT)/bin/findtoolchain wp76xx prefix) \
-	MANGOH_BOARD=yellow \
-	mksys -t wp76xx $(MKSYS_ARGS_COMMON) $(MKSYS_ARGS_YELLOW) mangOH.sdef
+LEGATO ?= 1
+.PHONY: legato_%
+legato_%:
+	$(eval export LEGATO_TARGET := $(subst legato_,,$@))
+ifeq ($(LEGATO),0)
+	echo "Not building LEGATO due to \$$LEGATO == 0"
+else
+	$(MAKE) -C $(LEGATO_ROOT) framework_$(LEGATO_TARGET)
+endif
 
-.PHONY: yellow_wp77xx
-yellow_wp77xx: legato_wp77xx
+# Build goals that get the target WP module type from the LEGATO_TARGET environment variable.
+.PHONY: $(BOARDS)
+$(BOARDS):
+ifndef LEGATO_TARGET
+	$(error LEGATO_TARGET not specified and no goal like $@_wp76xx provided)
+endif
+	if [ "$@" = "yellow" ] ; then $(call cyp_bld,$@_$(LEGATO_TARGET)) ; fi
```

Comment:
The YELLOW_GOALS are top-level goals themselves.  These are the old-style goals like 'yellow_wp76xx' that don't rely on LEGATO_TARGET being set.  That said, I can probably factor out some code here by making the $(BOARDS) goals use these other goals.

Reply:
The other reason I had this as a separate rule, rather than reusing the board_module rules is that when using leaf, you don't need to run findtoolchain to figure out what tool chain to use.  But, after doing some empirical testing on my build machine, I see it doesn't make a noticeable difference to build time, and they seem to be compatible with leaf.

Reply:
Take a look at the latest version and see if you like it better.

Reply:
Looks good